### PR TITLE
[WIP] Trying rxjs for a store

### PR DESCRIPTION
**Pull request by e-schultz**
@danielfigueiredo @igor-ka @winkerVSbecks - playing around with exposing a store as an observable stream using redux-rx. still WIP, also playing around to see how it works with change detection and various lifecycle hooks

### CODE REVIEW

**Comment by bennett000** [comment]
:+1:

**Comment by winkerVSbecks** [comment]
:+1: Wasn't aware of redux-rx, looks interesting.

**Comment by igor-ka** [comment]
:+1:

**Comment by e-schultz** [comment]
@bennett000 @igor-ka @danielfigueiredo @winkerVSbecks removed the redux-rx dependency since it needed rxjs 4, and made my own 'createObservableFromStore' using rx5. 

Also added some logging to onDestroy/onInit/etc to see how items are being created/destroyed based on state changes.

For example, when changing the filter from 'everyone' to 'alice' - you will see the items assigned to not-alice get destroyed, but all of the items owned by alice stay as they are. So not every part of the application is getting destroyed/re-created in response to state changes

**Comment by e-schultz** [comment]
@kosz @igor-ka @SethDavenport @bennett000 @AbdellaToronto @concertcoder 

Hey, take a look at StateService, and the containers/tasks/tasks.ts

But basically

```js
let owner$ = this.stateService.select(state => state.filters.get('owner'));
    let status$ = this.stateService.select(state=> state.filters.get('status'));
    let tasks$ = this.stateService.select(state=> state.tasks)
      .combineLatest(owner$, status$, (tasks, owner, status) => {
        return tasks.filter(n=> {
          const isDone = status === 'completed';
          return (n.get('done') === isDone || status === 'all')
            && (n.get('owner') === owner || owner === 'everyone')
        })
      }).subscribe(tasks=> this.tasks = tasks)
```

could do `this.task$ = ....` and drop off `.subscribe` and in the template just use the async pipe.

To dispatching actions:

```js
deleteTask = (task) => this.stateService.dispatch(TaskActions.deleteTask(task))
  loadTasks = () => this.stateService.dispatch(TaskActions.loadTasks())
  markTask =({task, newStatus}) => this.stateService.dispatch(TaskActions.markTask(task, newStatus))
```

A bit of tweaking could let you clean this up to

```js
deleteTask = (task) => this.stateService.dispatch(deleteTask(task))
  loadTasks = () => this.stateService.dispatch(loadTasks())
  markTask =({task, newStatus}) => this.stateService.dispatch(markTask(task, newStatus))
```

if you didn't like needing to do TaskActions.x

**Comment by igor-ka** [comment]
@e-schultz this looks good to me, especially after our meeting yesterday. :+1:

**Comment by danielfigueiredo** [comment]
:+1: Looks pretty good

**Comment by bennett000** [comment]
:+1:

File: app/src/containers/tasks/tasks.ts [hunk 1]
```diff
@@ -38,22 +41,32 @@ export default class Tasks implements OnDestroy, OnInit {
...
 
   ngOnInit() {
-    this.unsubscribe = this.ngRedux.connect(
-      this.mapStateToThis,
-      this.mapDispatchToThis
-    )(this);
+    
+
+
+    let owner$ = this.stateService.select(state => state.filters.get('owner'));
```
Comment (SethDavenport) [inline app/src/containers/tasks/tasks.ts]: `const` ?

**Comment by SethDavenport** [comment]
:+1:

**Comment by winkerVSbecks** [comment]
:+1:

**Comment by SethDavenport** [comment]
We sort of sold this already with ng2-redux, right?  Re-open if that's not true :)

**Pull request closed without merging**